acceptance/suites/tests/https/client_may_use_external_cert_chains.rb

```diff
@@ -0,0 +1,99 @@
+test_name "Ensure Puppet Server's HTTP client may use external cert chains" do
+
+  tmpdir = master.tmpdir('external-reports')
+  server_key = tmpdir + '/server.key'
+  server_cert = tmpdir + '/server.crt'
+  server_rb = tmpdir + '/server.rb'
+  directory_to_serve = tmpdir + '/public'
+
+  generate_self_signed_cert = "openssl req -new -newkey rsa:2048 -days 365 -nodes -x509 -keyout #{server_key} -out #{server_cert} -batch -addext 'subjectAltName = DNS:#{master}'"
+  run_server = "/opt/puppetlabs/puppet/bin/ruby #{server_rb} &"
+  kill_server = 'ps -ef | grep server.rb | grep -v grep | ruby -ne \'puts $_.split[1]\' | xargs -r kill' # `xargs -r` is a GNUism.
+  wait_for_server = "while [[ 0 -ne `curl -ksw '%{exitcode}' 'https://#{master}:7777/' | tail -1` ]]; do echo 'sleeping and waiting'; sleep 2; done"
+
+  server_script = <<EOF
+    require 'webrick'
+    require 'webrick/https'
+    require 'openssl'
+    require 'yaml'
+    require 'puppet'
+
+    cert = OpenSSL::X509::Certificate.new(File.read('#{server_cert}'))
+    key = OpenSSL::PKey.read(File.read('#{server_key}'))
```

Comment:
This is part of a HEREDOC (hence the `#{}` around `server_key`). String interpolation will not automatically wrap the string in quotes, so without the `'` this will create a script something like: 
```rb
key = OpenSSL::PKey.read(File.read(/tmp/external_ssl_x7vzadfy/server.key))
```

Reply:
ohhh, I see what you mean. 😅